query.go

```diff
@@ -24,11 +26,30 @@ func skippedQueryContext(_ context.Context, _ string, _ []driver.NamedValue) (dr
 	return nil, driver.ErrSkip
 }
 
+type attributeCtxKey struct{}
+
+func getAttributes(ctx context.Context) []attribute.KeyValue {
+	attrs, ok := ctx.Value(attributeCtxKey{}).([]attribute.KeyValue)
+	if !ok || attrs == nil {
+		attrs = make([]attribute.KeyValue, 0)
+	}
+	return attrs
+}
+
+// AddCutomAttributes adds additional OTel attributes to the context.
+// These attributes are added to metrics and spans generated for the query.
+func AddCustomAttributes(ctx context.Context, labels ...attribute.KeyValue) context.Context {
```

Comment:
```suggestion
func WithCustomAttributes(ctx context.Context, labels ...attribute.KeyValue) context.Context {
```

The naming could follow how https://pkg.go.dev/context#WithValue package does